share/csm_share/shr/shr_flux_mod.F90

```diff
@@ -355,6 +356,697 @@ real(R8)    :: psix2  ! stability function at ztref reference height
+   real(R8)    :: cp      ! specific heat of moist air
+   real(R8)    :: bn      ! exchange coef funct for interpolation
+   real(R8)    :: bh      ! exchange coef funct for interpolation
+   real(R8)    :: fac     ! vertical interpolation factor
+   real(R8)    :: DTiter  !              
+   real(R8)    :: DSiter  !              
+   real(R8)    :: DViter  !              
+
+   real(R8)    :: Dcool   !              
+   real(R8)    :: Qdel    ! net cool skin heating
+   real(R8)    :: Hd      ! net heating above -z=d             
+   real(R8)    :: Hb      ! net kinematic heating above -z = delta              
+   real(R8)    :: lambdaV !              
+   real(R8)    :: Fd      ! net fresh water forcing above -z=d
+   real(R8)    :: ustarw  ! surface wind forcing of layer above -z=d
+
+   real(R8)    :: Qsol   ! solar heat flux (W/m2)             
+   real(R8)    :: Qnsol  ! non-solar heat flux (W/m2) 
+   real(R8)    :: hour   !              
+   real(R8)    :: fsine  !              
+
+   real(R8)    :: SSS  ! sea surface salinity              
+   real(R8)    :: alphaT  !              
+   real(R8)    :: betaS  !              
+
+   real(R8)    :: doL     ! ocean forcing stablity parameter             
+   real(R8)    :: Rid     ! Richardson number at depth d              
+   real(R8)    :: Ribulk  ! Bulk  Richardson number at depth d
+   real(R8)    :: FofRi   ! Richardon number dependent diffusivity
+   real(R8)    :: Smult   ! multiplicative term based on regime              
+   real(R8)    :: Sfact   ! multiplicative term based on regime              
+   real(R8)    :: Kdiff   ! diffusive term based on regime              
+   real(R8)    :: Kvisc   ! viscosity term based on regime
+   real(R8)    :: hsign   !              
+   real(R8)    :: rhocn   !              
+   real(R8)    :: rcpocn  !              
+   real(R8)    :: Nreset  ! value for multiplicative reset factor             
+   real(R8)    :: resec   ! reset offset value in seconds              
+   logical     :: lmidnight
+   logical     :: ltwopm
+   logical     :: ltwoam
+   logical     :: lnoon
+   logical     :: lfullday
+   integer     :: nsum
+   integer     :: ier
+   real(R8)    :: pexp   ! eqn 19
+   real(R8)    :: AMP    ! eqn 18
+   real(R8)    :: dif3   
+   real(R8)    :: phid   
+   real(R8)    :: spval
+
+
+   !--- local functions --------------------------------
+   real(R8)    :: qsat   ! function: the saturation humididty of air (kg/m^3)
+   real(R8)    :: cdn    ! function: neutral drag coeff at 10m
+   real(R8)    :: psimhu ! function: unstable part of psimh
+   real(R8)    :: psixhu ! function: unstable part of psimx
+   real(R8)    :: Umps   ! dummy arg ~ wind velocity (m/s)
+   real(R8)    :: Tk     ! dummy arg ~ temperature (K)
+   real(R8)    :: xd     ! dummy arg ~ ?
+   real(R8)    :: molvisc ! molecular viscosity
+   real(R8)    :: molPr   ! molecular Prandtl number 
+
+
+   qsat(Tk)   = 640380.0_R8 / exp(5107.4_R8/Tk)
```

Comment:
Thanks for catching that.
On Jun 25, 2015 4:38 PM, "Sean Patrick Santos" <notifications@github.com>
wrote:

> In share/csm_share/shr/shr_flux_mod.F90
> <https://github.com/CESM-Development/cime/pull/33#discussion_r33312776>:
>
> > +   real(R8)    :: spval
> > +
> > +
> > +   !--- local functions --------------------------------
> > +   real(R8)    :: qsat   ! function: the saturation humididty of air (kg/m^3)
> > +   real(R8)    :: cdn    ! function: neutral drag coeff at 10m
> > +   real(R8)    :: psimhu ! function: unstable part of psimh
> > +   real(R8)    :: psixhu ! function: unstable part of psimx
> > +   real(R8)    :: Umps   ! dummy arg ~ wind velocity (m/s)
> > +   real(R8)    :: Tk     ! dummy arg ~ temperature (K)
> > +   real(R8)    :: xd     ! dummy arg ~ ?
> > +   real(R8)    :: molvisc ! molecular viscosity
> > +   real(R8)    :: molPr   ! molecular Prandtl number
> > +
> > +
> > +   qsat(Tk)   = 640380.0_R8 / exp(5107.4_R8/Tk)
>
> This is not a new issue by any stretch, but given that we've been trying
> to consolidate saturation calculations across CESM, it's unfortunate to
> have these one-off qsat implementations being not only kept but copied
> around.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/CESM-Development/cime/pull/33/files#r33312776>.
>